create_derivatives/routines.py

```diff
@@ -65,8 +67,104 @@ def get_as_uri(self, bag_filepath):
 
 
 class JP2Maker:
-    # TO DO: make JPG2000 derivatives
-    pass
+    """Creates JP2000 derivatives from TIFFs.
+
+    Creates JP2 directory in bag's data directory and JP2 derivatives.
+    TIFFs to be converted are targeted in the service directory. If the service directory is empty or
+    does not exist, target TIFFs at the root of the data directory.
+
+    Returns:
+        A tuple containing human-readable message along with list of bag identifiers.
+        Exceptions are raised for errors along the way.
+    """
+
+    def run(self):
+        bags_with_jp2s = []
+        for bag in Bag.objects.filter(process_status=Bag.PREPARED):
+            service_dir = Path(bag.bag_path, "data", "service")
+            if service_dir.is_dir() and len(service_dir):
+                tiff_files_dir = Path(bag.bag_path, "data", "service")
+            else:
+                tiff_files_dir = Path(bag.bag_path, "data")
+            self.jp2_list = self.create_jp2(bag, tiff_files_dir)
```

Comment:
I'm not sure you need to assign to a variable since you're not doing anything with it?